app/src/main/java/org/mozilla/reference/browser/Components.kt

```diff
@@ -133,4 +140,32 @@ class Components(
             init()
         }
     }
+
+    private val sentryService: SentryService by lazy {
+        SentryService(
+            applicationContext,
+            SENTRY_TOKEN,
+            sendEventForNativeCrashes = true
+        )
+    }
+    private val socorroService: MozillaSocorroService by lazy {
+        MozillaSocorroService(applicationContext, "ReferenceBrowser")
+    }
+
+    val crashReporter: CrashReporter by lazy {
+        CrashReporter(
+            services = listOf(sentryService, socorroService),
+            shouldPrompt = CrashReporter.Prompt.ALWAYS,
+            promptConfiguration = CrashReporter.PromptConfiguration(
+                appName = applicationContext.getString(R.string.app_name),
+                organizationName = "Mozilla"
+            ),
+            nonFatalCrashIntent = createNonFatalPendingIntent(),
```

Comment:
If we do not set the pending intent here then the library should just show the dialog. Can you try that? That may be easier and would require less code than showing the snackbar. :)

Reply:
I opted for refactoring the integration in its own class.
When I do not supply the nonFatalCrashIntent parameter, the app it not restarted properly. I recorded a small video.
https://drive.google.com/file/d/1vlsCtP4CVadFEI06Mgspt2OAjji3Cy80/view

Reply:
I'm trying to figure out, why any ideas ?

Reply:
For what I could find this is do to the app is taken from `BrowserActivity(MAIN)` -> `CrashReporterActivity (Dialog)` and when hit **"Restart Reference Browser"** `BrowserActivity(MAIN)` is getting relaunch but it was already in there, causing this weird state **(the black screen)**

Reply:
Can you file an issue for that in the a-c repository? Maybe we can get that fixed in the next sprint.

The reason why I wanted to avoid adding the custom UI here:
* It's from the sample and not a real in-app UI that we would build for crash reporting. And I think it's not worth building a real UI at this point. :)
* It adds strings and for the reason above we may not want to translate them if we are going to translate the app.
* It adds more code to the app :)

-> I'm fine with merging this as-is though; if without it the reporting is broken.

Reply:
👍